pattern_library/context_modifiers.py

```diff
@@ -0,0 +1,53 @@
+from collections import defaultdict
+from operator import attrgetter
+from typing import Callable
+
+from django.core.exceptions import ImproperlyConfigured
+
+from .cm_utils import accepts_kwarg
+
+GENERIC_CM_KEY = "__generic__"
+ORDER_ATTR_NAME = "__cm_order"
+
+
+class ContextModifierRegistry(defaultdict):
+    def __init__(self):
+        super().__init__(list)
+
+    def register(self, func: Callable, template: str = None, order: int = 0) -> None:
+        """
+        Adds a context modifier to the registry.
+        """
+        if not callable(func):
+            raise ImproperlyConfigured(
+                f"Context modifiers must be callables. {func} is a {type(func).__name__}."
+            )
+        if not accepts_kwarg(func, "context"):
+            raise ImproperlyConfigured(
+                f"Context modifiers must accept a 'context' argument. {func} does not."
```

Comment:
Nitpick: we could make this more specific and say "keyword argument" rather than just "argument".

Reply:
Implemented in 8eafff0